pkg/sanity/backup_restore.go

```diff
@@ -0,0 +1,423 @@
+/*
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"strings"
+	"time"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/client"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+
+	"github.com/libopenstorage/openstorage/volume"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Volume [Backup Restore Tests]", func() {
+	var (
+		restClient   *client.Client
+		volumedriver volume.VolumeDriver
+		credentials  *api.CredCreateRequest
+		credUUID     string
+	)
+
+	BeforeEach(func() {
+
+		if cloudBackupConfig == nil {
+			Skip("Skipping cloud backup/restore tests")
+		}
+		var err error
+		restClient, err = volumeclient.NewDriverClient(osdAddress, volumeDriver, volume.APIVersion, "")
+
+		Expect(err).ToNot(HaveOccurred())
+		volumedriver = volumeclient.VolumeDriver(restClient)
+
+		By("Creating Credentials first")
+
+		if cloudBackupConfig != nil {
+			cloudParamsMap := cloudBackupConfig.CloudProviders["azure"]
```

Comment:
This is an Azure test, but line 33 does not specify that it is an azure test. Either add `Azure` to the description in line 33 and copy it for each aws, and gce, or make the key in 55 configurable.

Reply:
Made it such a way that , it will do cloud backup restore tests for all credentials in the config file , provided the credentials get validated